### PR TITLE
Add cloudstack extra mounts for bootstrap

### PR DESCRIPTION
**Pull request by wongni**
*Issue #, if available:*

*Description of changes:*
I added extra host paths mounted to CLI container: https://github.com/aws/eks-anywhere/pull/2097 However, these paths are not mounted to the bootstrap kind cluster so the capc controller container cannot mount the host paths related to the custom root CA certificates. 

*Testing (if applicable):*
Manually tested

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/cloudstack/decoder/decoder.go [hunk 1]
```diff
@@ -61,6 +62,21 @@ func ParseCloudStackSecret() (*CloudStackExecConfig, error) {
 	}, nil
 }
 
+func GetCloudStackExtraMounts() ([]string, error) {
+	var extraMounts []string
+	env, found := os.LookupEnv(EksaCloudStackHostPathToMount)
```
Comment (maxdrib) [inline pkg/providers/cloudstack/decoder/decoder.go:67]: it seems like this should be a list object on the cloudstackdatacenterconfig, no? Could you remind me once more why we decided to go with the env var route?

File: pkg/providers/cloudstack/cloudstack.go [hunk 2]
```diff
@@ -242,7 +242,16 @@ func (p *cloudstackProvider) UpdateKubeConfig(_ *[]byte, _ string) error {
 }
 
 func (p *cloudstackProvider) BootstrapClusterOpts() ([]bootstrapper.BootstrapClusterOption, error) {
-	return common.BootstrapClusterOpts(p.datacenterConfig.Spec.ManagementApiEndpoint, p.clusterConfig)
+	extraMounts, err := decoder.GetCloudStackExtraMounts()
```
Comment (maxdrib) [inline pkg/providers/cloudstack/cloudstack.go:245]: could we add some unit tests here?

File: pkg/executables/kind.go [hunk 3]
```diff
@@ -135,6 +136,22 @@ func (k *Kind) WithExtraDockerMounts() bootstrapper.BootstrapClusterClientOption
 	}
 }
 
+func (k *Kind) WithExtraCloudStackMounts(extraMounts []string) bootstrapper.BootstrapClusterClientOption {
```
Comment (maxdrib) [inline pkg/executables/kind.go:139]: could we add some unit tests here?

File: cmd/eksctl-anywhere/cmd/createcluster.go [hunk 4]
```diff
@@ -221,16 +220,11 @@ func (cc *createClusterOptions) directoriesToMount(clusterSpec *cluster.Spec, cl
...
-			mountDirs := strings.Split(env, ",")
-			for _, dir := range mountDirs {
-				if _, err := os.Stat(dir); err != nil {
-					return nil, fmt.Errorf("invalid host path to mount: %v", err)
-				}
-				dirs = append(dirs, dir)
-			}
+		extraMounts, err := decoder.GetCloudStackExtraMounts()
+		if err != nil {
+			return nil, fmt.Errorf("invalid host path to mount: %v", err)
```
Comment (maxdrib) [inline cmd/eksctl-anywhere/cmd/createcluster.go:225]: this error message may be confusing. We're not doing any validation on the host path itself, right? I feel like this more represents an issue with retrieving the extra mounts

File: pkg/bootstrapper/bootstrapper.go [hunk 5]
```diff
@@ -134,6 +135,12 @@ func WithExtraPortMappings(ports []int) BootstrapClusterOption {
 	}
 }
 
+func WithExtraCloudStackMounts(extraMounts []string) BootstrapClusterOption {
```
Comment (maxdrib) [inline pkg/bootstrapper/bootstrapper.go:138]: it seems like a unit test here should be straightforward, no?

File: pkg/providers/cloudstack/decoder/decoder.go (hunk 1 above)
Comment (wongni) [reply pkg/providers/cloudstack/decoder/decoder.go:67]: Check this @vivek-koppuru 's comment about the env var approach: https://github.com/aws/eks-anywhere/pull/1981#issuecomment-1122626761

File: cmd/eksctl-anywhere/cmd/createcluster.go (hunk 4 above)
Comment (wongni) [reply cmd/eksctl-anywhere/cmd/createcluster.go:225]: GetCloudStackExtraMounts function checks if the host paths are existing. If one of the host paths doesn't exist the function returns an error.

File: pkg/executables/kind.go [hunk 6]
```diff
@@ -50,6 +50,7 @@ type kindExecConfig struct {
 	RegistryCACertPath     string
 	ExtraPortMappings      []int
 	DockerExtraMounts      bool
+	CloudStackExtraMounts  []string
```
Comment (chrisdoherty4) [inline pkg/executables/kind.go:53]: Why does Kind have something Cloud Stack specific? Kind is just a development cluster management tool with some quirks around how it works. It has no reason to know about Cloud Stack.

File: pkg/executables/kind.go [hunk 7]
```diff
@@ -135,6 +136,22 @@ func (k *Kind) WithExtraDockerMounts() bootstrapper.BootstrapClusterClientOption
...
 }
 
+func (k *Kind) WithExtraCloudStackMounts(extraMounts []string) bootstrapper.BootstrapClusterClientOption {
+	return func() error {
+		if k.execConfig == nil {
+			return errors.New("kind exec config is not ready")
+		}
+
+		if len(extraMounts) == 0 {
+			return errors.New("no extraMounts found in the list")
```
Comment (chrisdoherty4) [inline pkg/executables/kind.go:146]: `extraMounts` does mean anything to someone reading the error. Rule of thumb, don't include variable names in errors.

With respect to this behavior, why does it matter that no mounts were specified - wouldn't that just be a noop?

**Pull request closed without merging**